keps/sig-network/4631-loadbalancerstatus/README.md

```diff
@@ -0,0 +1,1066 @@
+  preservation, but these tests do not care about that:
+
+    - [`LoadBalancers should not have connectivity disruption during
+      rolling update with externalTrafficPolicy=Local`](https://github.com/kubernetes/kubernetes/blob/a6b08a8ea4/test/e2e/network/loadbalancer.go#L980).
+
+    - [`LoadBalancers ExternalTrafficPolicy: Local should only target
+      nodes with endpoints`](https://github.com/kubernetes/kubernetes/blob/a6b08a8ea4/test/e2e/network/loadbalancer.go#L1077).
+
+    - [`LoadBalancers ExternalTrafficPolicy: Local should target all
+      nodes with endpoints`](https://github.com/kubernetes/kubernetes/blob/a6b08a8ea4/test/e2e/network/loadbalancer.go#L1153).
+
+- These tests using `ExternalTrafficPolicy: Local` need to be split
+  into two tests each: one that just tests connectivity, and allows a
+  degraded load balancer (as above); and one that tests source IP
+  preservation as well, and thus uses `RequiredLoadBalancerFeatures:
+  ["ExternalTrafficPolicy"]`.
+
+    - [`LoadBalancers ExternalTrafficPolicy: Local should work for
+      type=LoadBalancer`](https://github.com/kubernetes/kubernetes/blob/a6b08a8ea4/test/e2e/network/loadbalancer.go#L1012).
+
+    - [`LoadBalancers ExternalTrafficPolicy: Local should work from pods`](https://github.com/kubernetes/kubernetes/blob/a6b08a8ea4/test/e2e/network/loadbalancer.go#L1200).
+
+- These tests will use `RequiredLoadBalancerFeatures: ["SessionAffinity"]`
+  and may skip themselves if the cloud provider indicates that the
+  feature is unsupported. They do not test source IP preservation and
+  thus will not set `"ExternalTrafficPolicy"` as a required feature.
+
+    - [`LoadBalancers should have session affinity work for LoadBalancer
+      service with ( Local | Cluster ) traffic
+      policy`](https://github.com/kubernetes/kubernetes/blob/a6b08a8ea4/test/e2e/network/loadbalancer.go#L565).
+
+    - [`LoadBalancers should be able to switch session affinity for
+      LoadBalancer service with ( Local | Cluster ) traffic
+      policy`](https://github.com/kubernetes/kubernetes/blob/a6b08a8ea4/test/e2e/network/loadbalancer.go#L575).
+
+- This test will use `RequiredLoadBalancerFeatures: ["LoadBalancerSourceRanges"]`
+  and may skip itself if the cloud provider indicates that the feature
+  is unsupported.
+
+    - [`LoadBalancers should only allow access from service loadbalancer
+      source ranges`](https://github.com/kubernetes/kubernetes/blob/a6b08a8ea4/test/e2e/network/loadbalancer.go#L422).
+
+- We lack tests of dual-stack and multi-protocol load balancer
+  provisioning, and should add those now that it's possible for clouds
+  to signal their failure to support it.
+
+- We should add a test for "cloud providers ignore Services with
+  `LoadBalancerClass` set".
+
+### Graduation Criteria
+
+#### Alpha
+
+- Helpers for the new API implemented in `k8s.io/cloud-provider`
+
+- cloud-provider-kind and cloud-provider-gcp set conditions
```

Comment:
I don't think there is people today to work with cloud-provider-gcp and review the changes, so this is not likely going to happen and cloud-provider-kind can not be the only thing implementing it

Reply:
oh... hm, that's sort of annoying if true since it means we can't really test this fully within k8s CI. We can change this to say AWS or Azure instead, but who will actually end up testing it in that case?

Reply:
this is the problem with Service loadbalancer, the last changes added to status, like IPMode and the relax validation on multiple TCP and UDP ports (it seems only Azure) were never implemented and they are not likely going to be implemented so I don't know if this is worth the effort ... in theory Gateway API is the things should be replacing this Service type